rclcpp/src/rclcpp/duration.cpp

```diff
@@ -226,4 +226,10 @@ Duration::seconds() const
   return std::chrono::duration<double>(std::chrono::nanoseconds(rcl_duration_.nanoseconds)).count();
 }
 
+Duration
+duration_from_seconds(double seconds)
```

Comment:
What do you think about making this a static method on the `Duration` class itself? `rclcpp::Duration::from_seconds(...);` is clear to me than what's in your tests at the moment.

Reply:
ROS1 also had a Time/Duration constructor of the signature:

`explicit TimeBase(double t)`
`explicit DurationBase(double t)`

Is there any reason that we couldn't include them here?

Reply:
Well, with the constructors you have no hint (other than docs) what the units of the double is. At least with `from_seconds(double)` you know it's seconds.

I'd be ok with something like `explicit Duration(std::chrono::nanoseconds)` and `explicit Duration(std::chrono::duration<double>)`, or even:

```c++
template<ChronoDurationT>
Duration(ChronoDurationT duration)
: storage(std::chrono::duration_cast<std::chrono::nanoseconds>(duration).count())
{}
```
(would need more than that, but something like that would get you started)

Reply:
I was preferring the option with the two separate constructors for `nanoseconds` and `duration<double>`. That way I could keep template complexity out of the header, but it fails when I do `Duration(1s)`. Overload resolution is ambiguous in that case, so I guess I'll go with the template version.

Do you prefer to define the type `ChronoDurationT` in the class or outside the class?

I'm going to go ahead and add the static `from_seconds` as well. I think that works better when the time is a variable instead of a constant. eg
```c++
double t = 1.0;
auto d = Duration::from_seconds(t);  // with static from_seconds method
Duration d(std::chrono::duration<double>(t));  // without from_seconds
```

Reply:
Yeah, `chrono` can be tricky sometimes, we do something like this here:

- https://github.com/ros2/rclcpp/blob/144c24c8fd15d57382fa994c7ecf00b847b1c339/rclcpp/include/rclcpp/executor.hpp#L162-L166

But I'm not sure if that would let you have two constructors, one for `int64_t` and one for `double` or not. We then forward that to the non-templated one that is always nanoseconds (which can also be implemented in the `.cpp`):

- https://github.com/ros2/rclcpp/blob/144c24c8fd15d57382fa994c7ecf00b847b1c339/rclcpp/include/rclcpp/executor.hpp#L282-L286

I don't mind about the name of the template argument, `TimeT` would also be fine I guess. We don't see to have a consistent convention unfortunately.

Thanks for iterating.

Reply:
That might work if the disambiguation rules prefer matching any integer type over double. However, it still brings template baggage into the header. I hate to do that. It is so beautifully simple right now :-) I'll get something working soon and update the PR for further feedback